doc-src/dev.rst

```diff
@@ -0,0 +1,369 @@
+   :code:`--system` argument to install in :code:`/usr/local` instead. (This
+   matches previous :code:`pip` behavior.) See Debian bugs `725848
+   <https://bugs.debian.org/725848>`_ and `820856
+   <https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=820856>`_.
+
+Publishing to the web
+~~~~~~~~~~~~~~~~~~~~~
+
+If you have write access to the repository, you can update the web
+documentation (i.e., http://hpc.github.io/charliecloud).
+
+Normally, :code:`doc` is normal directory ignored by Git. To publish to the
+web, that diretory needs to contain a Git checkout of the :code:`gh-pages`
+branch (not a submodule). To set that up::
+
+  $ rm -Rf doc
+  $ git clone git@github.com:hpc/charliecloud.git doc
+  $ cd doc
+  $ git checkout gh-pages
+
+To publish::
+
+  $ make web
+
+It sometimes takes a few minutes for the web pages to update.
+
+
+Coding style
+============
+
+We haven't written down a comprehensive style guide. Generally, follow the
+style of the surrounding code, think in rectangles rather than lines, and
+avoid CamelCase.
+
+Note that Reid is very picky about style, so don’t feel singled out if he
+complains (or even updates this section based on your patch!). He tries to be
+nice about it.
+
+Writing English
+---------------
+
+* When describing what something does (e.g., your PR or a command), use the
+  `imperative mood <https://chris.beams.io/posts/git-commit/#imperative>`_,
+  i.e., write the orders you are giving rather than describe what the thing
+  does. For example, do:
+
+    | Inject files from the host into an image directory.
+    | Add :code:`--join-pid` option to :code:`ch-run`.
+
+  Do not (indicative mood):
+
+    | Injects files from the host into an image directory.
+    | Adds :code:`--join-pid` option to :code:`ch-run`.
+
+* Use sentence case for titles, not title case.
+
+* If it's not a sentence, start with a lower-case character.
+
+
+:code:`curl` vs. :code:`wget`
+-----------------------------
+
+For URL downloading in shell code, including Dockerfiles, use :code:`wget -nv`.
+
+Both work fine for our purposes, and we need to use one or the other
+consistently. According to Debian's popularity contest, 99.88% of reporting
+systems have :code:`wget` installed, vs. about 44% for :code:`curl`. On the
+other hand, :code:`curl` is in the minimal install of CentOS 7 while
+:code:`wget` is not.
+
+For now, Reid just picked :code:`wget` because he likes it better.
+
+Variable conventions in shell scripts and :code:`.bats` files
+-------------------------------------------------------------
+
+* Separate words with underscores.
+
+* User-configured environment variables: all uppercase, :code:`CH_TEST_`
+  prefix. Do not use in individual :code:`.bats` files; instead, provide an
+  intermediate variable.
+
+* Variables local to a given file: lower case, no prefix.
+
+* Bats: set in :code:`common.bash` and then used in :code:`.bats` files: lower
+  case, :code:`cht_` prefix.
```

Comment:
They are currently set with the `ch_` prefix, exceptions being `ch_timg` and `ch_ttar` for chtests.  FWIW, I prefer the `ch_` prefix.